src/html/HTMLWriter.jl

```diff
@@ -930,6 +930,17 @@ function render_head(ctx, navnode)
         RD.fontawesome_css...,
         RD.katex_css,
     ]
+    collapse_script = ""
+    if get(getpage(ctx, navnode).globals.meta, :DocStringsCollapsed, false)
+        # if DocStringsCollapse = true in `@meta`, we let JavaScript click the
+        # collapse button after that page has loaded.
+        collapse_script = script("""
+        window.onload = function() {
+         var toggleButton = document.getElementById('documenter-article-toggle-button');
+         if (toggleButton) {toggleButton.click();}
+        };
+        """)
```

Comment:
I feel that ideally we'd do this by statically generating the necessary HTML. It looks like we use `display: none` on the `section` tags.. and then update some other stuff. Ideally, we'd have a class on the `article` tag that toggles everything.

How much effort would you be willing to put into this? I didn't test this yet, but I assume that this works, and we can merge this as is, and refactor later as well.

Reply:
> More importantly, rendering the page expanded and then collapsing it with JavaScript seems like the right thing to do from an accessibility standpoint: We'd want the page to still be useful without JavaScript.

Ah.. that is a fair point actually. We should indeed make sure that it loads uncollapsed if you don't have JS enabled.

Reply:
Right. I'd still advocate for going with the approach in this PR. I started out with trying to write out the collapsed HTML if the option is set until I realized that:

(a) It's *much* more difficult to implement that way. In fact, I'm not sure that I'd be able to do it in any reasonable amount of time.

(b) "Collapsed" means `display: none`, which *removes* the content entirely without JS, which seems like a bad idea.

The test in the PR only checks for the presence of the injected JS. I don't think there's much I can do beyond that. I did test locally that the JS does what it's supposed to (very nicely, in fact). What I could do as an additional test is to *use* the feature in the Documenter documentation. Maybe collapsing all the [Internal API](https://documenter.juliadocs.org/stable/lib/internals/anchors/) pages by default?

Reply:
Yeah, let's go with the JS approach.

It currently does have the problem that you can see all the docstrings collapsing when you navigate to the page though. Maybe @Hetarth02 has some ideas how to work around that.

Reply:
Interesting, I will definitely take a look into this.

Reply:
@Hetarth02 I'd appreciate that! You might be better at JS than me. I'm having real trouble getting it to load correctly. In my first iteration, it worked fine locally but only 90% of the time in Chrome and only 1% of the time in Safari. I'm guessing the script was running too early. In the current iteration, I've tried to move the `<script>` tag to the end of the file, but that's still behaving the same way (i.e., not working in Safari). I also tried putting it in an external script (I think local scripts load before external scripts, which may be part of the problem), but that was much more complicated to implement, and then just didn't work at all. I can push that branch again somewhere if you'd like to take a look at it.

Reply:
> It currently does have the problem that you can see all the docstrings collapsing when you navigate to the page though

Yeah, I can't decide whether it's a problem or whether I actually kinda like it. But in any case, I think jQuery is handling the collapse animation. Maybe there's a way to temporarily disable the animation or at least speed it up?

Reply:
Yes, Jquery handles the animation and there is a way to speed it up.

I also think that, initially appearing as open then getting closed is okay.

Reply:
> I can push that branch again

Pushed to https://github.com/JuliaDocs/Documenter.jl/tree/mg/meta-docstrings-collapsed2. This one doesn't work at all, though, and I'm not sure why.

Reply:
@goerz Which branch should I check from?

`mg/meta-docstrings-collapsed2` or `mg/meta-docstrings-collapsed`

Reply:
Preferably, `mg/meta-docstrings-collapsed`. The other one was an attempt to put the JS in a separate script file. It was both more complicated and didn't actually work, but I just wanted to keep the branch around in case `mg/meta-docstrings-collapsed` doesn't work out and I need to get back to it.

Reply:
`mg/meta-docstrings-collapsed` works perfectly when looking at documentation locally with `LiveServer`, but doesn't work reliably when looking at the online documentation, e.g. at https://documenter.juliadocs.org/previews/PR2394/lib/internals/anchors/. In Chrome, it works most of the time, in Safari, it does not work most of the time. Presumably, the issue is the asynchronous execution of JS and the code the clicks the collapse button firing too early.

Reply:
@goerz So, I fiddled around a bit. I have found a way which I think will work.

Basically, what we will do is save all the `@meta` flags in `data` attribute of a `div` element. Something like below,

```html
<div data-docstringscollapsed="true"></div>
```

Then add a `js` script, which will perform the necessary actions for each flags. For example, like this

```js
// assets/html/js/metadata.js

// libraries: jquery
// arguments: $

let docstringscollapsed = $("div[data-docstringscollapsed]").data();

if (docstringscollapsed) {
  $("#documenter-article-toggle-button").click();
}
```

Doing this will always ensure that, all the scripts and DOM content will be loaded.

Reply:
That sounds good! Do you want to just push to the `mg/meta-docstrings-collapsed` branch? I'm assuming you have permissions.

Reply:
I don't think I have the necessary permissions to do so.

Reply:
I can merge from https://github.com/Hetarth02/Documenter.jl/ into my PR then. Let me know which branch.

Reply:
Alright, will do so. Let me iron some things out and will ping you when I am done.

Reply:
No worries! Thanks for taking a look! I'll check it out.

Reply:
@goerz Just FYI, this is just a rudimentary implementation from my side. There is of course room for improvement in logic.😅

Reply:
It works perfectly! Thanks for coming up with the approach that makes this reliable!